prow/clonerefs/run.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clonerefs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"sync"
+
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/pod-utils/clone"
+)
+
+// Run clones the configured refs
+func (o *Options) Run() error {
```

Comment:
Any thoughts on making this `Run(o)`?
What does `Run()` write to `o`? If nothing thoughts on making this `Options` instead of `*Options`?

Reply:
The `Validate()` `Complete()` and `Run()` pattern was taken from `kubectl`. We should be read-only on config so not having a pointer receiver would be fine, whether it's a method or a function I don't feel strongly.